### DISCUSSION TITLE
auth realtime update

### DISCUSSION THREAD

**Discussion by thal-sh** [pocketbase/pocketbase#4850] [Q&A] [answered]
I'm making a client side app and I wanna use auth, when the user login I want the btn to update immediately to logout and vise versa, do I use auth realtime? I'm using normal email and password auth

**Comment by ganigeorgiev** [comment, marked as answer]
There are no realtime APIs for this (and I don't think there is need for it).

PocketBase is fully stateless and doesn't have sessions in the traditional sense (even the tokens are not stored in the database) and the auth state is maintained locally as part of the `pb.authStore` when using the SDKs (that is also why there is no "logout" method but `pb.authStore.clear()`).

To track auth store changes:

- For the JS SDK you can register a [`pb.authStore.onChange((token, model) => {})` callback function](https://github.com/pocketbase/js-sdk?tab=readme-ov-file#common-auth-store-fields-and-methods).
- For the Dart SDK you can use the [`pb.authStore.onChange` stream](https://github.com/pocketbase/dart-sdk?tab=readme-ov-file#authstore).

**Reply by thal-sh** [reply]
is this normal js sdk or I need to expand it to a framework

**Reply by ganigeorgiev** [reply]
I don't understand what you are asking. The above links are functionality that is available as part of the [JS SDK](https://github.com/pocketbase/js-sdk) and [Dart SDK](https://github.com/pocketbase/dart-sdk).

**Reply by thal-sh** [reply]
`  useEffect(() => {

    const removeListener = pb.authStore.onChange((token) => {
        if(token) {
          setIsLoggedIn(true);
        }else {
          setIsLoggedIn(false)
        }
    });

    return () => {
        removeListener();
    };
}, []);`
is this good?
